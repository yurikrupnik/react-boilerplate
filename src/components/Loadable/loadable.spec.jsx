import React from 'react';
import Component, { Loading } from './index';

const {
    describe,
    it,
    expect,
    shallow,
} = global;

describe(`${Component.name}`, () => {
    it('render', () => {

        const wrapper = shallow(
            <Component />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('render loading with timeOut', () => {

        const wrapper = shallow(
            <Loading timedOut={true} />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('render loading with error', () => {
        const wrapper = shallow(
            <Loading error={{}} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('render loading with retry', () => {

        const wrapper = shallow(
            <Loading retry={() => {}} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('render loading with pastDelay', () => {

        const wrapper = shallow(
            <Loading pastDelay={true} />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
