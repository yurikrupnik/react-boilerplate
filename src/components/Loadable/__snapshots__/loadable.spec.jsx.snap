// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_default render 1`] = `<Component />`;

exports[`_default render loading with error 1`] = `
<div>
  <div>
    Error!
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Retry
  </button>
</div>
`;

exports[`_default render loading with pastDelay 1`] = `
<div>
  <div>
    Error!
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Retry
  </button>
</div>
`;

exports[`_default render loading with retry 1`] = `
<div>
  <div>
    Error!
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Retry
  </button>
</div>
`;

exports[`_default render loading with timeOut 1`] = `
<div>
  <div>
    Error!
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Retry
  </button>
</div>
`;
